### PR TITLE
Update tutorials for nk3

### PR DESCRIPTION
**Pull request by PhilipVinc**

### CODE REVIEW

File: Tutorials/Heisenberg1d.ipynb [hunk 1]
```diff
@@ -138,70 +140,108 @@
...
+    "    variational_state=vs)\n",
     "\n",
     "start = time.time()\n",
-    "gs.run(output_prefix='Jastrow', n_iter=300)\n",
+    "gs.run(300, out='Jastrow')\n",
     "end = time.time()\n",
     "\n",
     "print('### Jastrow calculation')\n",
-    "print('Has',ma.n_par,'parameters')\n",
+    "print('Has',nk.jax.tree_size(vs.parameters),'parameters')\n",
```
Comment (gcarleo) [inline Tutorials/Heisenberg1d.ipynb:231]: I was wondering if we could somehow still have some easy to use ```n_par``` for models? maybe ```nk.models.n_par()``` ?

File: Tutorials/Heisenberg1d.ipynb [hunk 2]
```diff
@@ -276,53 +316,63 @@
...
+      "100%|██████████| 600/600 [00:17<00:00, 33.77it/s, Energy=-39.123 ± 0.014 [σ²=0.204, R̂=0.9995]]\n"
+     ]
+    },
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
       "### RBM calculation\n",
       "Has 528 parameters\n",
-      "The RBM calculation took 79.31553912162781 seconds\n"
```
Comment (gcarleo) [inline Tutorials/Heisenberg1d.ipynb:288]: oh yeah that's a good time improvement :)

File: Tutorials/Heisenberg1d.ipynb (hunk 1 above)
Comment (PhilipVinc) [reply Tutorials/Heisenberg1d.ipynb:231]: I can put it in Variational State and will internally call what you see above.

File: Tutorials/jax.ipynb [hunk 3]
```diff
@@ -173,23 +184,23 @@
...
+      "0 5.36 ± 0.15 [σ²=15.31, R̂=0.9999]\n",
+      "50 -2.92862 ± 0.00027 [σ²=0.00000, R̂=1.4135]\n",
+      "100 -2.975733 ± 0.000024 [σ²=0.000000, R̂=1.4135]\n",
+      "150 -2.979545 ± 0.000018 [σ²=0.000000, R̂=1.4135]\n",
+      "200 -2.983247 ± 0.000017 [σ²=0.000000, R̂=1.4135]\n",
+      "250 -2.991311 ± 0.000056 [σ²=0.000000, R̂=1.4135]\n",
+      "300 -2.991311 ± 0.000056 [σ²=0.000000, R̂=1.4135]\n",
+      "350 -2.991311 ± 0.000056 [σ²=0.000000, R̂=1.4135]\n",
+      "400 -2.994530 ± 0.000011 [σ²=0.000000, R̂=1.4135]\n",
+      "450 -2.994530 ± 0.000011 [σ²=0.000000, R̂=1.4135]\n"
```
Comment (gcarleo) [inline Tutorials/jax.ipynb:203]: why the outputs are so different with respect to the old tutorial?

File: Tutorials/Heisenberg1d.ipynb (hunk 1 above)
Comment (PhilipVinc) [reply Tutorials/Heisenberg1d.ipynb:231]: Addressed by commit in main branch

File: Tutorials/jax.ipynb (hunk 3 above)
Comment (PhilipVinc) [reply Tutorials/jax.ipynb]: they aren't anymore.
You were using a complex network before and i had a real one.
+ the initialisation thing

**Pull request merged**